src/RTG/ChatFilter/Loader.php

```diff
@@ -33,64 +33,41 @@
 use pocketmine\event\player\PlayerChatEvent;
 
 class Loader extends PluginBase implements Listener {
-
-    public $words;
+    public $cfg;
 
     public function onEnable() {
-        $this->getServer()->getPluginManager()->registerEvents($this, $this);
-        $this->words = array();
+        $this->getServer()->getPluginManager()->registerEvents($this, $this)
         $this->cfg = new Config($this->getDataFolder() . "words.txt", Config::ENUM);
-
-        $this->words = $this->cfg->getAll();
-    }
-
-    public function onSave() {
-        $this->cfg = new Config($this->getDataFolder() . "words.txt", Config::ENUM);
-        $this->cfg->setAll($this->words);
-        $this->cfg->save();
     }
 
     public function onCommand(CommandSender $sender, Command $command, $label, array $args) {
         switch (strtolower($command->getName())) {
-
             case "cf":
                 if ($sender->hasPermission("chatfilter.command") or $sender->isOp()) {
-
                     if (isset($args[0])) {
                         switch (strtolower($args[0])) {
-
                             case "add":
-
                                 if (isset($args[1])) {
-
                                     $word = $args[1];
-
-                                    if (isset($this->words[strtolower($word)])) {
-
+                                    if ($this->cfg->exists($word, true)) {
                                         $sender->sendMessage("[ChatFilter] $word exists!");
                                     } else {
-
-                                        $this->words[strtolower($word)] = strtolower($word);
+                                        $this->cfg->set(strtolower($word));
```

Comment:
$this->cfg->set($k) adds an entry $k to an enum config.

#blameshoghi - I don't nknow why he wants to put entirely different implementations of Config (they don't even share anything in common except that they are configs!)